### PR TITLE
os/arch/xtensa: remove spiflash include dependency

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
1. move the header files from external/esp_idf_port to os/arch to avoid
include dependency of external in os/arch.
2. modify the include path in c files accordingly
@sunghan-chang

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [2a74e60059a06d2935b3fd785dee9873ddef1947] - Code Rule Check Result:
os/arch/xtensa/src/esp32/rom/queue.h:136: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
os/arch/xtensa/src/esp32/rom/queue.h:271: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/arch/xtensa/src/esp32/rom/queue.h:306: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/xtensa/src/esp32/rom/queue.h:351: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/xtensa/src/esp32/rom/queue.h:351: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/arch/xtensa/src/esp32/rom/queue.h:352: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/xtensa/src/esp32/rom/queue.h:352: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/xtensa/src/esp32/rom/queue.h:353: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/xtensa/src/esp32/rom/queue.h:354: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/xtensa/src/esp32/rom/queue.h:390: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/xtensa/src/esp32/rom/queue.h:390: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/xtensa/src/esp32/rom/queue.h:454: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/xtensa/src/esp32/rom/queue.h:504: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/xtensa/src/esp32/rom/queue.h:504: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/xtensa/src/esp32/rom/queue.h:569: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/xtensa/src/esp32/rom/queue.h:628: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [2a74e60059a06d2935b3fd785dee9873ddef1947] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [819f0ec8f272af84dd02b86475fcf9e14dbaefd9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [819f0ec8f272af84dd02b86475fcf9e14dbaefd9] - Code Rule Check Result:
os/arch/xtensa/src/esp32/rom/queue.h:454: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/xtensa/src/esp32/rom/queue.h:504: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/xtensa/src/esp32/rom/queue.h:569: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/xtensa/src/esp32/rom/queue.h:628: ERROR: [IDT_M_TAB] please, no space before tabs

File: os/arch/xtensa/src/esp32/rom/queue.h [hunk 1]
```diff
@@ -303,7 +303,7 @@ struct {								\
 #define	STAILQ_LAST(head, type, field)					\
 	(STAILQ_EMPTY((head)) ?						\
 		NULL :							\
-	        ((struct type *)(void *)				\
+			((struct type *)(void *)				\
```
Comment (zhouxinhe) [inline os/arch/xtensa/src/esp32/rom/queue.h]: Two `Tab` are enough here.

File: os/arch/xtensa/src/esp32/rom/queue.h [hunk 2]
```diff
@@ -387,7 +387,7 @@ struct {								\
 	if (LIST_NEXT((elm), field) != NULL &&				\
 	    LIST_NEXT((elm), field)->field.le_prev !=			\
 	     &((elm)->field.le_next))					\
-	     	panic("Bad link elm %p next->prev != elm", (elm));	\
+			panic("Bad link elm %p next->prev != elm", (elm));	\
```
Comment (zhouxinhe) [inline os/arch/xtensa/src/esp32/rom/queue.h]: same above

File: os/arch/xtensa/src/esp32/esp_clk.h [hunk 3]
```diff
@@ -0,0 +1,93 @@
+/******************************************************************
```
Comment (zhouxinhe) [inline os/arch/xtensa/src/esp32/esp_clk.h]: rename `external/esp_idf_port/include/esp_clk.h` to `os/arch/xtensa/src/esp32/esp_clk.h` ?
or add a duplicated file?

**Comment by seinfra** [comment]
Target : [2a639087f665c927574394ef30e6baa1711d3c2e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2a639087f665c927574394ef30e6baa1711d3c2e] - Code Rule Check (C++) OK.

File: os/arch/xtensa/src/esp32/esp_clk.h [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+/**
+ * @brief Update the calibration value of RTC slow clock
+ *
+ * The value has to be in the same format as returned by rtc_clk_cal (microseconds,
+ * in Q13.19 fixed-point format).
+ * This value is used by timekeeping functions (such as gettimeofday) to
+ * calculate current time based on RTC counter value.
+ * @param value calibration value obtained using rtc_clk_cal
+ */
+void esp_clk_slowclk_cal_set(uint32_t value);
```
Comment (sunghan-chang) [inline os/arch/xtensa/src/esp32/esp_clk.h]: Let's include <stdint.h> for uint32_t and uint64_t

**Comment by seinfra** [comment]
Target : [947708e4003f6ce4545faa1a3119d75b75e1fcc0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [947708e4003f6ce4545faa1a3119d75b75e1fcc0] - Code Rule Check OK.

File: os/arch/xtensa/src/esp32/esp_clk.h (hunk 3 above)
Comment (SRCN-LiQin) [reply os/arch/xtensa/src/esp32/esp_clk.h]: add a duplicated file. this PR mainly focus on remove the unreasonable dependency from os/arch to external,

File: os/arch/xtensa/src/esp32/esp_clk.h (hunk 4 above)
Comment (SRCN-LiQin) [reply os/arch/xtensa/src/esp32/esp_clk.h]: fixed, thanks for suggestion

File: os/arch/xtensa/src/esp32/rom/queue.h (hunk 1 above)
Comment (SRCN-LiQin) [reply os/arch/xtensa/src/esp32/rom/queue.h]: fixed, thanks

**Comment by seinfra** [comment]
Target : [3379f598f841f6ea5c4ec9781ccc7448092c183a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3379f598f841f6ea5c4ec9781ccc7448092c183a] - Code Rule Check (C++) OK.

**Pull request merged**